### PR TITLE
fix: Mark standalone None as is_optional.

### PR DESCRIPTION
**Pull request by DanCardin**
## Description

This is not generally meaningful, because something is not really ever going to be annotated as `foo: None`.

However, when writing code that operates over something like `for t in TypeView(int | None).inner_types`, it can come up.

Perhaps in concert with https://github.com/litestar-org/type-lens/pull/10, I might be able to _avoid_ it, but... (it only occurred to me in writing this description :P), it maybe seems like the logical extension of:
```python
assert TypeView(str | int | None).is_optional is True
assert TypeView(int | None).is_optional is True
assert TypeView(None).is_optional is True
```

### CODE REVIEW

File: type_lens/type_view.py [hunk 1]
```diff
@@ -121,11 +121,16 @@ def is_subclass_of(self, cl: type[Any] | tuple[type[Any], ...]) -> bool:
             if self.origin in UNION_TYPES:
                 return all(t.is_subclass_of(cl) for t in self.inner_types)
 
-            return self.origin not in UNION_TYPES and issubclass(self.origin, cl)
+            return self.origin not in UNION_TYPES and isinstance(self.origin, type) and issubclass(self.origin, cl)
```
Comment (peterschutt) [inline type_lens/type_view.py:124]: Is this relevant to the PR, or accidental?

**Review by peterschutt — APPROVED** [review]
Thanks, makes sense to me!

I think the collection stuff is unrelated though.

File: type_lens/type_view.py (hunk 1 above)
Comment (DanCardin) [reply type_lens/type_view.py:124]: bah! sorry yes. Messed up my rebase for https://github.com/litestar-org/type-lens/pull/9

**Comment by peterschutt** [comment]
I'm having second thoughts on this, `Optional` is specifically a union that includes `None` in its members. Would an `is_none_type` property be better?

**Comment by DanCardin** [comment]
I have an `is_none_type` branch pending also. As i mentioned above the `strip_optional()` branch would also likely handle this case for my particular usecase. Thus, if you want to close this, i think i'll be fine.

But I was just thinking of the logical extension of the types. `None` is a special type that indicates "optional"ity, whether it's actually required or not. So i dont necessarily see the difference between `int | None` and `None | None` and `None` in terms of the expected result of this function, versus having the presense of `None` cause it to be `True`, but only when combined with other types.
